# Titanic: Aprendizaje Automático del Desastre


### Resumen

Los datos han sido divididos en 2 grupos:

training set (train.csv)
test set (test.csv)

El conjunto de entrenamiento deberá ser utilizado para construir el/los modelo/s de aprendizaje automático. Para el conjunto de entrenamiento, se dará el resultado (también conocido como "verdad básica") para cada pasajero. Su modelo, deberá estar basado en "características" como el géneros o clase de los pasajeros. También puedes jugar con algunas características para crear las tuyas propias.

El conjunto de prueba debe usarse para ver qué tan bien se desempeña su modelo en datos sin clasificar. Para el conjunto de pruebas, no proporcionamos la verdad básica para cada pasajero. Debes poder predecir estos resultados. Para cada pasajero en el conjunto de pruebas, use el modelo que entrenó para predecir si sobrevivieron o no al hundimiento del Titanic.

También incluimos gender_submission.csv, un conjunto de predicciones que asumen que todas las pasajeros y solo las mujeres sobreviven, como ejemplo de cómo debería ser un archivo de envío

# Parte 1: Análisis de Exploratorio de Datos

En esta parte se hace un análisis exploratorio de datos para determinar: 

- ¿Qué variables se presentan en el conjunto de datos?
- ¿Qué tipo de variables posee el conjunto de datos?
- ¿Existe alguna relación dichas variables?
- ¿Poseen las variables del conjunto de entrenamiento y el conjunto de prueba las mismas características?
- ¿Qué atributos del conjunto de entrenamiento tienen una mayor implicancia sobre la variable final de decisión?

In [11]:
# Importing Libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# Opening Datasets
train_data = pd.read_csv('data/train.csv')
test_data = pd.read_csv('data/test.csv')

In [3]:
print('El conjunto de datos de entrenamiento (training) posee ', train_data.shape[0], ' filas y ', train_data.shape[1], ' columnas.')
print('El conjunto de datos de prueba (testing) posee ', test_data.shape[0], ' filas y ', test_data.shape[1], ' columnas.')

El conjunto de datos de entrenamiento (training) posee  891  filas y  12  columnas.
El conjunto de datos de prueba (testing) posee  418  filas y  11  columnas.


### Obs.

El conjunto de datos de entrenamiento posee un atributo más que el conjunto de datos de testing, que viene siendo la variable de decisión (sobrevive o no sobrevive). Además el conjunto de datos de entrenamiento posee más observaciones que el conjunto de datos de testing (891 contra 418).

Ahora se procede a comparar los conjuntos de datos de entrenamiento y prueba para ver si sus variables numéricas poseen características númericas similares (como por ejemplo media, varianza, cuartiles, etc.)

In [23]:
train_data[['Pclass','Age','SibSp','Parch','Fare']].describe()

Pclass         Age       SibSp       Parch        Fare
count  891.000000  714.000000  891.000000  891.000000  891.000000
mean     2.308642   29.699118    0.523008    0.381594   32.204208
std      0.836071   14.526497    1.102743    0.806057   49.693429
min      1.000000    0.420000    0.000000    0.000000    0.000000
25%      2.000000   20.125000    0.000000    0.000000    7.910400
50%      3.000000   28.000000    0.000000    0.000000   14.454200
75%      3.000000   38.000000    1.000000    0.000000   31.000000
max      3.000000   80.000000    8.000000    6.000000  512.329200

In [24]:
test_data[['Pclass','Age','SibSp','Parch','Fare']].describe()

Pclass         Age       SibSp       Parch        Fare
count  418.000000  332.000000  418.000000  418.000000  417.000000
mean     2.265550   30.272590    0.447368    0.392344   35.627188
std      0.841838   14.181209    0.896760    0.981429   55.907576
min      1.000000    0.170000    0.000000    0.000000    0.000000
25%      1.000000   21.000000    0.000000    0.000000    7.895800
50%      3.000000   27.000000    0.000000    0.000000   14.454200
75%      3.000000   39.000000    1.000000    0.000000   31.500000
max      3.000000   76.000000    8.000000    9.000000  512.329200

### Obs.

Ambos conjuntos de datos no poseen grandes diferencias en variables numéricas, por ende, se puede trabajar con ambos utilizando los mismos criterios.